### DISCUSSION TITLE
How to use ExcelJs in the ClientSide please tutorial

### DISCUSSION THREAD

**Discussion by ghost** [exceljs/exceljs#2496] [Q&A]
Please do a tutorial, how to implement the library for Front-End, I am new to this and I have tried everything without any results.
It is very confusing for me. And I think for others too, thank you very much!

**Comment by cncolder** [comment]
Did you means work on browser?

Here is an example for export excel and download.

```ts
import ExcelJS from 'exceljs/dist/es5/exceljs.browser.js'
import { saveAs } from 'file-saver'

async function saveAsExcel() {
    const wb = new ExcelJS.Workbook()

    const ws = wb.addWorksheet()

    const row = ws.addRow(['a', 'b', 'c'])
    row.font = { bold: true }

    const buf = await wb.xlsx.writeBuffer()

    saveAs(new Blob([buf]), 'abc.xlsx')
}
```

**Comment by lae0901** [comment]
I try that and get error:
Uncaught SyntaxError: The requested module './node_modules/exceljs/dist/es5/exceljs.browser.js' does not provide an export named 'default'

`<script src="module.mjs" type="module" ></script>

import ExcelJS from './node_modules/exceljs/dist/es5/exceljs.browser.js' ;

`

**Comment by bleuscyther** [comment]
```typescript
import * as Excel from "exceljs/dist/exceljs.min.js";
 ...
 let workbook = new Excel.Workbook();
```

**Comment by i365** [comment]
Here is a simple demo for working on client web browser
[exceljs example demo for browser ](https://jstool.gitlab.io/demo/exceljs/?g)

**Comment by anshumansworld** [comment]
> Did you means work on browser?
> 
> Here is an example for export excel and download.
> 
> ```ts
> import ExcelJS from 'exceljs/dist/es5/exceljs.browser.js'
> import { saveAs } from 'file-saver'
> 
> async function saveAsExcel() {
>     const wb = new ExcelJS.Workbook()
> 
>     const ws = wb.addWorksheet()
> 
>     const row = ws.addRow(['a', 'b', 'c'])
>     row.font = { bold: true }
> 
>     const buf = await wb.xlsx.writeBuffer()
> 
>     saveAs(new Blob([buf]), 'abc.xlsx')
> }
> ```

```
let workbook = new Excel.Workbook();
...
...
workbook.xlsx.writeBuffer().then((buf) => {
  saveAs(new Blob([buf]), `testFile.xlsx`);
  // other stuffs
});
```

It's working fine on local but nothing happens on stage env after deployment. This is happening when I use npm package and importing minified version as:
`import * as Excel from 'exceljs/dist/exceljs.min';`

it's working fine if I use
`import * as Excel from 'exceljs/dist/exceljs';`

**Comment by AkshayRawatt** [comment]
Hi , i am using the same thing , but making a http clinet req for fetching excel template files with conditional formatting and styles.
after downloading the new sheet with values in it, not able to retain the styles. Please helppppppppp SOS anyone???????``

here's  the  code snippet.

    const workbook = new Excel.Workbook(
     { useStyles: true}
  )
    this._http.get(FILE_PATH, { responseType: 'arraybuffer' }).subscribe((res: any) => {
      const arryBuffer = new Response(res).arrayBuffer();
      arryBuffer.then(function (data) {
        workbook.xlsx.load(data)
          .then(function () {
            
//data modification values being set

            workbook.xlsx.writeBuffer().then((data) => {
              let blob = new Blob([data], { type: EXCEL_TYPE });
              fs.saveAs(blob, fileName + FILE_EXTENSION);
            });
          });
      });
    });

**Comment by RoWEN-FCUB** [comment]
> { useStyles: true}

I'm having a similar problem:

```
exportLocalXlsx(index: number): void {
    const workBook: Workbook = new Workbook();
    const subscription = this.http.get('./assets/Modelo.xlsx', { responseType: 'blob' })
    .subscribe(value => {
    const blob: Blob = value;
    const reader = new FileReader();
    const tplan = this.totalPlan;
    const ac_plan = this.erecords[index].planacumulado;
    const ac_real = this.erecords[index].realacumulado;
    const d_plan = this.erecords[index].plan;
    const d_real = this.erecords[index].consumo;
    const edate = this.erecords[index].fecha.toString().substr(0, this.erecords[index].fecha.toString().indexOf('T'));
    reader.onload = function (e: any) {
      const contents = e.target.result;
      console.log(contents);
      workBook.xlsx.load(contents).then(data => {
        // workBook.getWorksheet(1).addRow(['new data', 1, 2, 3]);
        // const valor = workBook.worksheets[0].getCell(1, 1).value;
        // console.log(workBook.worksheets[0].getCell(2, 1).style);
        workBook.worksheets[0].getCell(3, 14).value = tplan;
        workBook.worksheets[0].getCell(3, 15).value = ac_plan;
        workBook.worksheets[0].getCell(3, 16).value = ac_real;
        workBook.worksheets[0].getCell(3, 17).model.result = undefined;
        workBook.worksheets[0].getCell(3, 18).model.result = undefined;
        workBook.worksheets[0].getCell(3, 19).value = d_plan;
        workBook.worksheets[0].getCell(3, 20).value = d_real;
        workBook.worksheets[0].getCell(3, 21).model.result = undefined;
        workBook.worksheets[0].getCell(3, 22).model.result = undefined;
        workBook.xlsx.writeBuffer().then(data => {
          const blobUpdate = new Blob([data], { type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet' });
          fsaver.saveAs(blobUpdate, 'Modelo 5 - ' + edate + '.xlsx');
        });
      });
    };
    reader.readAsArrayBuffer(blob);
    });
  }
```
After I download the file, it looses all styles, and Microsoft Excel reports that it has removed the styles xml. Any updates on this?

**Comment by RoWEN-FCUB** [comment]
After a few test I find that the problem was the excel file used, not my code. Now it's working as spected.

**Comment by OmkarKumbare** [comment]
Used exceljs for implementation

Here is simple example for export excel file using react [StackBlitz](https://stackblitz.com/edit/react-export-to-excel?file=src%2FApp.js)

**Reply by LogiMateenSA** [reply]
works nicely :)

**Comment by SevenZark** [comment]
> 

I installed file-saver and copy/pasted this exact code. In Chrome, this results in 'Failed - No file.'

**Comment by Ahmed-2125** [comment]
I imported xlsx file in my web app using ExcelJs , I get styles as OpenXml e.g( font , fill{pattern:..,indexed:64 },color:{theme:6} ),Im unable to convert it into styled object for my react app ,please share information( medium ) or good example  Thank you.

**Comment by UV-99** [comment]
i am don
get js link from here https://nexinfotech.com/
